### PR TITLE
Release 0.2.0

### PR DESCRIPTION
**Pull request by MichaelAquilina**
* Adds testing for multiple versions of ZSH
* Fixes bugs around BESTMATCH and ordering of results
* Handle multiple options for BESTMATCH by choosing alias with shortest length
* Change default mode to BESTMATCH

### CODE REVIEW

File: tests/test_aliases.zunit [hunk 1]
```diff
@@ -47,3 +47,27 @@
     assert "$output" same_as '\033[1mFound existing alias for "git status -v". You should use: "gsv"\033[00m'
     assert $state equals 0
 }
+
+@test 'aliases bestmatch - choosest shortest alias key on equal value lengths' {
```
Comment (MichaelAquilina) [inline tests/test_aliases.zunit]: typo - choosest

File: tests/test_aliases.zunit [hunk 2]
```diff
@@ -37,7 +37,7 @@
     assert $state equals 0
 }
 
-@test 'aliases match - bestmatch' {
+@test 'aliases bestmatch - choosest longest matching alias value' {
```
Comment (MichaelAquilina) [inline tests/test_aliases.zunit]: typo - choosest

File: you-should-use.plugin.zsh [hunk 3]
```diff
@@ -89,22 +89,26 @@ function _check_aliases() {
...
 
-      found_aliases+="$k"
+        found_aliases+="$k"
 
-      if [[ "${#v}" -gt "${#best_match}" ]]; then
-        best_match="$k"
+        # Match aliases to longest portion of command
+        if [[ "${#v}" -gt "${#best_match}" ]]; then
+            best_match="$k"
+          # on equal length, choose the shortest alias
```
Comment (MichaelAquilina) [inline you-should-use.plugin.zsh]: misalignment

**Comment by MichaelAquilina** [comment]
Try use (is_substring_of)[https://github.com/molovo/zunit/pull/64] assert function if zunit 0.8.0 releases before next release

**Comment by MichaelAquilina** [comment]
@molovo not sure if you're interested, but I am looking for someone to review my code before I merge it. Usually I get different people to do it and since I interacted with you last thought it would be worth a try. Feel free not to if you dont feel like though :)

**Comment by molovo** [comment]
@MichaelAquilina sure, I'll take a look

File: .travis.yml [hunk 4]
```diff
@@ -1,17 +1,24 @@
...
+ - ZSH_VERSION=5.1.1 URL=https://downloads.sourceforge.net/project/zsh/zsh/5.1.1/zsh-5.1.1.tar.xz
+ - ZSH_VERSION=5.0.8 URL=https://downloads.sourceforge.net/project/zsh/zsh/5.0.8/zsh-5.0.8.tar.gz
+
 addons:
   apt:
     packages:
     - build-essential
 
 before_install:
     - export LOCAL="$(mktemp --directory --tmpdir=${TMPDIR:/tmp} local.bin.XXXXXX)"
```
Comment (molovo) [inline .travis.yml]: Might be worth caching this directory to speed up builds

File: tests/test_aliases.zunit [hunk 5]
```diff
@@ -47,3 +47,27 @@
...
+    assert "$output" same_as '\033[1mFound existing alias for "vim". You should use: "v"\033[00m'
+    assert $state equals 0
+}
+
+@test 'aliases exact match' {
+    alias hu='hg update'
+    run _check_aliases 'hg update'
+
+    assert "$output" same_as '\033[1mFound existing alias for "hg update". You should use: "hu"\033[00m'
+}
```
Comment (molovo) [inline tests/test_aliases.zunit]: Forgot to check `$state` here

File: tests/test_aliases.zunit [hunk 6]
```diff
@@ -47,3 +47,27 @@
...
+
+@test 'aliases substring match' {
+    alias hu='hg update'
+    run _check_aliases 'hg update -all'
+
+    assert "$output" same_as '\033[1mFound existing alias for "hg update". You should use: "hu"\033[00m'
```
Comment (molovo) [inline tests/test_aliases.zunit]: Forgot to check `$state` here

**Comment by MichaelAquilina** [comment]
Thanks for the review @molovo, let me know if I can help out with anything!

**Comment by molovo** [comment]
@MichaelAquilina just trying this out locally, and noticed that occasionally `you-should-use` will report twice for different aliases when a shorter one exists.

screen shot 2017-07-05 at 12 07 53

Is this expected?

**Comment by molovo** [comment]
In fact, managed to get it to report three times here:

screen shot 2017-07-05 at 12 13 29

Want me to raise an issue?

**Comment by MichaelAquilina** [comment]
by default the package shows all matched results. Have you set `export YSU_MODE=BEST`?

**Comment by molovo** [comment]
@MichaelAquilina Ah! Knew I was missing something. Thanks

**Comment by MichaelAquilina** [comment]
No problem! It tells me that I should probably make this option more visible or possible rethink the default since you are not the first one to have this problem

**Pull request merged**